Add string methods: `isupper()` and `islower()`

balapriyac opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

Add Python string methods `islower()` and `isupper()`. Closes #387 

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry


### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

yangc95 commented:
Hi @balapriyac! 👋 Review 1️⃣ will be done shortly

yangc95 commented on content/python/concepts/strings/terms/islower/islower.md:
```diff
@@ -0,0 +1,44 @@
+  - 'paths/analyze-data-with-python'
+---
+
+The `.islower()` string method takes in a string and returns `True` if all the letters in the string are in lowercase, else returns `False`. This method ignores spaces, newlines, numeric and special characters in the string.
```
```suggestion
The `.islower()` string method takes in a string and returns `True` if all the letters in the string are in lowercase, else returns `False`. This method ignores spaces, newlines, numeric, and special characters in the string.
```

yangc95 commented on content/python/concepts/strings/terms/isupper/isupper.md:
````diff
@@ -0,0 +1,44 @@
+print(my_string.isupper())
+# Output: False
+```
+
````
```suggestion
```

yangc95 commented on content/python/concepts/strings/terms/isupper/isupper.md:
```diff
@@ -0,0 +1,44 @@
+  - 'paths/analyze-data-with-python'
+---
+
+The `.isupper()` string method takes in a string and returns `True` if all the letters in the string are in uppercase, else returns `False`. This method ignores spaces, newlines, numeric and special characters in the string.
```
```suggestion
The `.isupper()` string method takes in a string and returns `True` if all the letters in the string are in uppercase, else returns `False`. This method ignores spaces, newlines, numeric, and special characters in the string.
```

yangc95 reviewed (approved):
❗ I see no real faults with either of these entries. @balapriyac I just suggested some missing commas and removed a space.

Going to update those and pass this along.

SSwiniarski commented:
@balapriyac, I'll be doing review 2.

SSwiniarski reviewed (approved):
It all looks good to me.

Dusch4593 commented:
@balapriyac @SSwiniarski @yangc95 about to merge but made changes in 3e15c203a9726dfb3beecfb58b167bb78abaec82 and 78ea81875338584f653e28a0e3c0f44e89524aaf where I added some text in the `Codebyte Example` sections.

yangc95 commented:
@balapriyac We strongly encourage you to add a profile picture to your Codecademy profile so the entry has a visible contributor (although, if you can jog my memory, did you wish to stay anonymous?) ☺️ 

<img width="899" alt="Screen Shot 2021-12-27 at 4 14 50 PM" src="https://user-images.githubusercontent.com/72277593/147507726-0cd88e74-09f8-4da6-bca4-f0f67da8d435.png">

Your entries are live here for reference:

`islower()`:
https://www.codecademy.com/resources/docs/python/strings/islower
`isupper()`:
https://www.codecademy.com/resources/docs/python/strings/isupper

balapriyac commented:
Hi @yangc95, I had updated my Codecademy profile earlier, and have just added my GitHub username as well.
I'm not sure why it's showing `anonymous`. I guess it will take a while to show up on the docs page? 😊

yangc95 commented:
> Hi @yangc95, I had updated my Codecademy profile earlier, and have just added my GitHub username as well. I'm not sure why it's showing `anonymous`. I guess it will take a while to show up on the docs page? 😊

I hope that's the case! Though I'm wondering why I don't see your handle on the list of community contributors either. How does your privacy settings look?

<img width="608" alt="Screen Shot 2021-12-28 at 2 25 36 PM" src="https://user-images.githubusercontent.com/72277593/147600232-3029322f-3b03-4e68-bfe2-04b44f80bbb2.png">

yangc95 commented:
@balapriyac checking back in ⏯️ any development?

balapriyac commented:
Hi Christine, I'm not sure why I'm still `anonymous`. I'm able to view that
my Codecademy profile is visible to everyone (in my settings).

Earlier this year, when my hacktoberfest PRs were merged, I was able to see
my GitHub username in the docs that went live. Now, even those have the
tag, `Anonymous contributor`.

I'm not sure how this works. Is it like—I've updated my GitHub profile link
in my Codecademy profile. So only my subsequent PRs will be attributed to
me?

I thought maybe we should try that the next time and see if it's getting
updated. 😄

Happy new year to you. 🎉💫



On Thu, 30 Dec 2021, 8:29 pm Christine Yang, ***@***.***>
wrote:

> @balapriyac <https://github.com/balapriyac> checking back in ⏯️ any
> development?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/477#issuecomment-1003065301>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALIW4EYOEGR3JTVK4ZAJU3LUTRXWDANCNFSM5KNRYSUA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Dusch4593 commented:
@balapriyac There may be some internal logic issue regarding how contributors are displayed on the entries. My colleagues and I just id'd another recently merged PR with the same issue as you. I just passed all this info along to my contacts in the Engineering team and we'll hopefully get some more insight into what's going on. I don't expect any updates until next week after the holidays are done, though. But I really appreciate your bringing this to our attention as we want you to get the credit you deserve!

balapriyac commented:
Thank you very much, Brandon. Yes, I hadn't quite noticed this. It was
Christine who had identified this, and asked me if I wished to remain
anonymous as the contributor name wasn't showing up—that's how I got to
know of this. 🙂

Thanks again. We can revisit this early next year. No problem at all.

On Thu, 30 Dec 2021, 8:56 pm Brandon Dusch, ***@***.***>
wrote:

> @balapriyac <https://github.com/balapriyac> There may be some internal
> logic issue regarding how contributors are displayed on the entries. My
> colleagues and I just id'd another recently merged PR with the same issue
> as you. I just passed all this info along to my contacts in the Engineering
> team and we'll hopefully get some more insight into what's going on. I
> don't expect any updates until next week after the holidays are done,
> though. But I really appreciate your bringing this to our attention as we
> want you to get the credit you deserve!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/477#issuecomment-1003075952>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALIW4EZJ626T6Y7MYR2V36DUTR23VANCNFSM5KNRYSUA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Dusch4593 commented:
Happy New Year @balapriyac! We have an update!

In addition to setting your Codecademy profile visibility to "Everyone", you have to enable SAML single sign-on ([SSO](https://docs.github.com/en/enterprise-cloud@latest/authentication/authenticating-with-saml-single-sign-on/about-authentication-with-saml-single-sign-on)) between your Codecademy and GitHub profiles _through_ the same primary email address. This can be done by following some steps [here in the CONTRIBUTING.md guide](https://github.com/Codecademy/docs/blob/main/.github/CONTRIBUTING.md#codecademy-username).

Hope this helps but definitely let us know if you have any questions! 😃

balapriyac commented:
Thank you very much @Dusch4593. I'll do that soon. 🙂

balapriyac commented:
![image](https://user-images.githubusercontent.com/47279635/148685651-2d8bfba7-b516-4a07-99a1-d7f4684b0113.png)
Thanks for the help @yangc95 and @Dusch4593, I'm now able to see my name beside my contributions.😄
